### ISSUE TITLE
Cannot load some of STS pre-trained adapters

### ISSUE THREAD

**Issue by sasaadi** [Adapter-Hub/adapter-transformers#137]
I am having issues regarding loading the sts/sts-b@ukp with heads adapter:
1) on BERT
```
model = BertForSequenceClassification.from_pretrained("bert-base-uncased")
config = AdapterConfig.load("houlsby")  # also with Pfeiffer config
model.load_adapter("sts/sts-b@ukp", "text_task", config=config)
```
which raises the following error:

```
Error(s) in loading state_dict for BertForSequenceClassification:
        size mismatch for classifier.weight: copying a param with shape torch.Size([1, 768]) from checkpoint, the shape in current model is torch.Size([2, 768]).
        size mismatch for classifier.bias: copying a param with shape torch.Size([1]) from checkpoint, the shape in current model is torch.Size([2]).
```
and when I replace `BertForSequenceClassification` with `BertModelWithHeads`, it does not include the prediction head on top.

2) on DistilBERT
```
model = DistilBertModelWithHeads.from_pretrained("distilbert-base-uncased")
config = AdapterConfig.load("pfeiffer")
model.load_adapter("sts/sts-b@ukp", "text_task", config=config)
```
which gives the following error:

`OSError: Unable to load weights from pytorch checkpoint file.`

I can produce the same error when I load the adapter from a local directory.

Everything works fine when I load adapters for the RoBERTa model.

Do you think whether there is a version issue? 

Thanks

## Environment info
- `transformers` version: 1.1.1
- Platform: Linux-4.15.0-126-generic-x86_64-with-glibc2.10
- Python version: 3.8.8
- PyTorch version (GPU?): 1.4.0 (True)

**Comment by calpt** [comment]
Hi @sasaadi,

The first error is due to a class label mismatch between the base Transformer model and the loaded adapter. `BertForSequenceClassification` is initialized with a binary classification head by default. Changing your code to the following should work:

```python
model = BertForSequenceClassification.from_pretrained("bert-base-uncased", num_labels=1)  # set num_labels to 1 for sts-b
config = AdapterConfig.load("houlsby")
model.load_adapter("sts/sts-b@ukp", "text_task", config=config)
```

Regarding the second error, I'm unable to reproduce this using PyTorch 1.7.1. However, similar issues in the HuggingFace repo ([here](https://github.com/huggingface/transformers/issues/4336#issuecomment-766996640) or [here](https://github.com/huggingface/transformers/issues/6620#issuecomment-722023621)) could be solved by upgrading to a newer PyTorch version.

**Comment by calpt** [comment]
This issue should be resolved with v2.1.x, where every should be loadable via `AutoModelWithHeads`. See [notes](https://github.com/Adapter-Hub/adapter-transformers/releases/tag/adapters2.1.0).

Closing this for now, please reopen if this issue still persists.
